components/fhir-core/src/main/java/org/wso2/healthcare/codegen/tool/framework/fhir/core/oas/OASGenerator.java

```diff
@@ -0,0 +1,378 @@
+                    propertySchema.$ref(APIDefinitionConstants.OAS_REF_SCHEMAS + type.getCode());
+                } else if (APIDefinitionConstants.DATA_TYPE_BACKBONE.equals(type.getCode())) {
+                    propertySchema.$ref(APIDefinitionConstants.OAS_REF_SCHEMAS + elementName);
+                } else {
+                    String oasDataType = OASGenUtils.mapToOASDataType(
+                            type.getCode().substring(type.getCode().lastIndexOf(".") + 1));
+                    propertySchema.setType(oasDataType);
+                    propertySchema.setPattern(OASGenUtils.getRegexForDataType(oasDataType));
+                }
+                propertySchema.setDescription(element.getDefinition());
+                propertySchemaMap.put(elementName.toString(), propertySchema);
+                if (element.getMin() != 0) {
+                    requiredElementsCollector.add(elementName.toString());
+                }
+            }
+            objectSchema.setProperties(propertySchemaMap);
+            allOfSchema.addAllOfItem(objectSchema);
+            List<String> requiredElements = new ArrayList<>(requiredElementsCollector);
+            allOfSchema.setRequired(requiredElements);
+            resourceOAS.getComponents().addSchemas(structureDefinition.getType(), allOfSchema);
+        }
+        return resourceOAS;
+    }
+
+    /**
+     * Populates OAS info object.
+     *
+     * @param apiDefinition API definition object
+     */
+    public void populateOASInfo(APIDefinition apiDefinition) {
+
+        Info info = new Info();
+        info.setTitle(apiDefinition.getResourceType());
+        info.setVersion("1.0.0");
+
+        //TODO: check whether these needs to be provided via config vars
+        License license = new License();
+        license.setName("Apache 2.0");
+        license.setUrl("https://www.apache.org/licenses/LICENSE-2.0.html");
+        Contact contact = new Contact();
+        contact.setName("API Support");
+        contact.setUrl("https://wso2.com/contact/`");
+        contact.setEmail("user@email.com");
+        info.setContact(contact);
+        apiDefinition.getOpenAPI().setInfo(info);
+    }
+
+    /**
+     * Populates OAS paths object.
+     *
+     * @param apiDefinition API definition object
+     */
+    private void populateOASPaths(APIDefinition apiDefinition) {
+
+        Paths paths = new Paths();
+        Map<String, String> interactions = new HashMap<>() {{
+            put("read", "GET");
+            put("search", "GET");
+            put("write", "POST");
+            put("update", "PUT");
+            put("delete", "DELETE");
+            put("patch", "PATCH");
+        }};
+
+        PathItem rootPath = new PathItem();
+        PathItem idPath = new PathItem();
+        for (Map.Entry<String, String> interaction : interactions.entrySet()) {
+
+            switch (interaction.getKey()) {
+                case "read":
+                    Operation readOp = new Operation();
```

Comment:
Can we move the Operation object creation outside of the switch/or to an util function, given it is common for all?